### DISCUSSION TITLE
How to allow Prisma migrations to continue working with custom index

### DISCUSSION THREAD

**Discussion by PaulBGD** [prisma/prisma#6677] [Q&A] [answered]
Hello,

In my application one thing I've needed was full text search with postgresql. Implementing this in the schema language seems "easy enough":

```
model MyModel {
  vectors   Unsupported("tsvector")?
}
```

However, in order to index this to gain better performance using a query like this:
```ts
db.$queryRaw(`CREATE INDEX IF NOT EXISTS idx_fts_model_vec ON "MyModel" USING gin(vectors)`)
```

Then every time I try to migrate I get the following error:
```
✔ Drift detected: Your database schema is not in sync with your migration history.
```

I assume the issue here is that the index is not defined in the schema, so Prisma assumes it has to completely rebuild the database. Is there some way to define a `gin` index in Prisma?

**Comment by jackbravo** [comment, marked as answer]
As stated by @tomhoule here: https://github.com/prisma/prisma/issues/7515#issuecomment-863051346 you can do this:

```
model MyModel {
  name String
  vectors   Unsupported("tsvector")?
  @@index([vector], name: "projects_vectors_idx")
}
```

Then generate the migration using `prisma migrate dev --create-only` and modify the index on the generated migration SQL file:

```diff
-CREATE INDEX "projects_vectors_idx" ON "MyModel"("vectors");
+CREATE INDEX "projects_vectors_idx" ON "MyModel" USING gin ("vectors");
```

I see that you can even add a function and a trigger:

```plsql
create or replace function vectorsCol() RETURNS trigger AS
$$
begin
    NEW."vectors" := to_tsvector(NEW."name");
    return NEW;
end;
$$ language plpgsql;

create trigger __200_vectors_col
    before insert or update on "MyModel"
    for each row
execute procedure vectorsCol();
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
